### PR TITLE
fix(dotenv): Empty string crashing parsing

### PR DESCRIPTION
**Pull request by lino-levan**
Empty strings in .env files crashed the parsing of them. Looks like it was just a falsey value that was not tested for.

Simple reproc (.env):
```
A=""
```

Related: #2798

### CODE REVIEW

File: dotenv/mod.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export function parse(
 
     env[key] = notInterpolated
       ? notInterpolated
-      : interpolated
+      : interpolated || interpolated === ""
```
Comment (cjihrig) [inline dotenv/mod.ts:60]: Could this become `typeof interpolated  === "string"`?

File: dotenv/testdata/.env.test [hunk 2]
```diff
@@ -5,6 +5,7 @@ AFTER_EMPTY=empty
 EMPTY_VALUE=
 QUOTED_SINGLE='single quoted'
 QUOTED_DOUBLE="double quoted"
+EMPTY_QUOTED=""
```
Comment (cjihrig) [inline dotenv/testdata/.env.test:8]: Should we include a test for empty single quotes as well?

File: dotenv/mod.ts (hunk 1 above)
Comment (lino-levan) [reply dotenv/mod.ts:60]: That's a lot more explicit, let's do that.

File: dotenv/testdata/.env.test (hunk 2 above)
Comment (lino-levan) [reply dotenv/testdata/.env.test:8]: I don't think it's strictly necessary but having one more test to prevent regressions will not hurt.

Comment (lino-levan) [reply dotenv/testdata/.env.test]: It turns out it was strictly necessary, there was a bug with single quote parsing. Nice catch!

**Review by cjihrig — APPROVED** [review]
LGTM, thanks!
